truffle/src/main/java/org/jruby/truffle/pack/runtime/PackFrame.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.pack.runtime;
+
+import com.oracle.truffle.api.frame.FrameDescriptor;
+import com.oracle.truffle.api.frame.FrameSlot;
+import com.oracle.truffle.api.frame.FrameSlotKind;
+
+public class PackFrame {
+
+    public static final PackFrame INSTANCE = new PackFrame();
```

Comment:
`PackFrame.INSTANCE.getSourceSlot()` looks a bit confusing to me.
Would it make sense to use static members to clarify why there is only one instance? (i.e. This is the *descriptor* of a Frame in PackNodes, not the frame itself)